impl/src/main/java/org/jboss/weld/bootstrap/BeanDeployment.java

```diff
@@ -279,4 +286,12 @@ public String toString() {
         return builder.toString();
     }
 
+    public void trim() {
+        for (SlimAnnotatedTypeContext<?> ctx : getBeanDeployer().getEnvironment().getAnnotatedTypes()) {
+            if (!AnnotatedTypes.hasBeanDefiningAnnotation(ctx.getAnnotatedType())) {
+                getBeanDeployer().getEnvironment().removeAnnotatedType(ctx);
```

Comment:
Then we should probably perform the "trim" operation "inside" the `BeanDeployerEnvironment`?